lib/pipeline.js

```diff
@@ -1658,6 +1658,75 @@ class PipelineModel extends BaseModel {
         return factory.list(listConfig);
     }
 
+    /**
+     * Fetch builds belonging to a pipeline or events
+     * @param  {Object}   [config]
+     * @param  {Number}   [config.sort]                         Sort rangekey by ascending or descending
+     * @param  {Object}   [config.paginate]                     Pagination parameters
+     * @param  {Number}   [config.paginate.count]               Number of items per page
+     * @param  {Number}   [config.paginate.page]                Specific page of the set to return
+     * @param  {Number}   [config.groupEventId]                 Group event ID
+     * @param  {Boolean}  [config.latest]                       If we want to return latest in groupEventId, default false
+     * @return {Promise}  Resolves to an array of builds
+     */
+    async getBuilds(config) {
+        const latest = hoek.reach(config, 'params.latest');
+        const groupEventId = hoek.reach(config, 'params.groupEventId');
+
+        // Fetch all builds for each event with same groupEventId
+        if (groupEventId && !latest) {
+            // Lazy load factory dependency to prevent circular dependency issues
+            // https://nodejs.org/api/modules.html#modules_cycles
+            /* eslint-disable global-require */
+            const EventFactory = require('./eventFactory');
+            /* eslint-enable global-require */
+            const eventFactory = EventFactory.getInstance();
+
+            const events = await eventFactory.list({
+                params: { groupEventId }
+            });
+
+            if (Array.isArray(events) && events.length > 0) {
+                const processed = [];
+
+                events.forEach(e => processed.push(e.getBuilds()));
+
+                return Promise.all(processed).then(builds => {
+                    // flatten array of arrays
+                    return builds.flat(1);
+                });
+            }
+
+            return events;
```

Comment:
is there a reason why we return events here instead of builds?

Reply:
It'll be an empty array anyways. so I can hardcode it to that if that makes more sense to you.

Actually since events should never be empty array I think I can remove that if section entirely.